Book list

BJordi opened the pull request:
## Summary

- Created BooksList component.
- Created BooksListItem.
- Added service to get all the books from the API.
- Added integration with books API and rendered its data.

## Screenshot

![image](https://user-images.githubusercontent.com/22583914/51282186-33ce8380-19c3-11e9-8080-764b3843f65e.png)

Commit:
Merge branch 'session-management' into book-list

Commit:
Books
- Added headers to api config.
- Added service to get all books.

Commit:
Renamed Auth view to NavBar.

Commit:
Books
- Added BookListItem component

Commit:
Books
- Added BookList component

Commit:
Books
- Added Dashboard component

Commit:
Books
- Added Dashboard to router

Commit:
Books
- Fixes in scoped classes

Commit:
BOOKS API INTEGRATION

- Added BooksListItem component.

Commit:
BOOKS API INTEGRATION

- Added BooksList component, that will use the BooksListItem component to render all the books obtained through the service interaction with Books API.

Commit:
BOOKS API INTEGRATION

Fixes in setting the API headers, as when using them after reloading the page,
the request throw a 401 error.

Commit:
BOOKS

- Removed unnecessary files.

Commit:
BOOKS API INTEGRATION

- Added call to the books API in the Dashboard component, in order to get all the books,
iterate over this books array and render all its items.

Commit:
Solved conflicts with master branch.

Commit:
REMOVED UNNECESSARY CODE AND CORRECT LINT ERRORS

Commit:
Solved conflicts with master branch.

FrankIglesias commented on src/services/books.js:
```diff
@@ -0,0 +1,6 @@
+import api, { API_DEFAULT_HEADERS } from '@/config/api.js'
+
+export const getAllBooks = () => {
+  api.setHeaders(API_DEFAULT_HEADERS)
```
This header must be setted before

BJordi replied:
Yeah, forgot to delete this! Thanks!

FrankIglesias commented on src/views/BooksListItem.vue:
```diff
@@ -0,0 +1,56 @@
+<template lang="pug">
+  .book-detail-container
+    img.book-image(:src='book.image_url')
```
add alt

FrankIglesias commented on src/views/BooksListItem.vue:
```diff
@@ -0,0 +1,56 @@
+    color: $black;
+    font-size: 20px;
+    font-weight: bold;
+    line-height: 1;
```
do you need this line-height: 1?

BJordi replied:
I will delete it.

FrankIglesias commented on src/views/BooksListItem.vue:
```diff
@@ -0,0 +1,56 @@
+  .subtitle {
+    font-size: 16px;
+    line-height: 1;
+    text-align: start;
```
left?

BJordi replied:
In this case is the same, but I will change it to left.

FrankIglesias commented on src/views/BooksListItem.vue:
```diff
@@ -0,0 +1,56 @@
+    text-align: start;
+  }
+}
+
```
delete this line

FrankIglesias commented on src/views/Dashboard.vue:
```diff
@@ -1,14 +1,35 @@
 <script>
 import NavBar from '@/components/NavBar'
+import BooksList from './BooksList'
+import { getAllBooks } from '../services/books.js'
```
@/services

Commit:
BOOKS API INTEGRATION: FIXES ASKED BY THE CODE REVIER

- Remove empty lines.
- Remove aliases in imports.
- Removed unnecessary line height.
- Replaced text-align value: 'start' -> 'left'
- Added alt attribute to img in book item.
- Removed api.setHeaders from service as it should be done before this. So, I added it in the api.config file.

Commit:
BOOKS API INTEGRATION

- Moved BooksList and BooksListItem from views folder to components one, as they weren't
in the right place according to the projects structure.
- Replaced relatives imports by aliases.

FrankIglesias commented on src/config/api.js:
```diff
@@ -1,10 +1,20 @@
 import { create } from 'apisauce'
+import LocalStorageService from '../services/localStorage'
```
@/services

Commit:
BOOKS API INTEGRATION: REPLACED RELATIVE IMPORT BY ALIASES

Commit:
BOOKS API INTEGRATION

- Added key to the v-for items.
- Fixed styles in the thumbnail image of the book. Now it appears cropped, but it used to be compressed.
